content/hardware/03.nano/boards/nano-matter/tutorials/matter-fan/content.md

```diff
@@ -0,0 +1,224 @@
+  }
+  while (!Matter.isDeviceCommissioned()) {
+    delay(200);
+  }
+
+  Serial.println("Waiting for Thread network...");
+  while (!Matter.isDeviceThreadConnected()) {
+    delay(200);
+    decommission_handler();
+  }
+  Serial.println("Connected to Thread network");
+
+  Serial.println("Waiting for Matter device discovery...");
+  while (!matter_fan.is_online()) {
+    delay(200);
+    decommission_handler();
+  }
+  Serial.println("Matter device is now online");
+}
+
+void loop()
+{
+  decommission_handler();
+  static uint8_t fan_last_speed = 0;
+  uint8_t fan_current_speed = matter_fan.get_percent();
+
+  if (fan_current_speed != fan_last_speed) {
+    fan_last_speed = fan_current_speed;
+    Serial.print("Fan speed: ");
+    Serial.print(fan_current_speed);
+    Serial.println("%");
+    int speed = map(fan_current_speed, 0, 100, 0, 255);
+    analogWrite(fan_pin, speed);
+  }
+
+  static bool fan_last_state = false;
+  bool fan_current_state = matter_fan.get_onoff();
+
+  if (fan_current_state != fan_last_state) {
+    fan_last_state = fan_current_state;
+    if (fan_current_state) {
+      Serial.println("Fan ON");
+    } else {
+      Serial.println("Fan OFF");
+    }
+  }
+}
+
+void decommission_handler()
+{
+  // If the button is not pressed or the device is not commissioned - return
+  if (digitalRead(BTN_BUILTIN) != LOW || !Matter.isDeviceCommissioned()) {
+    return;
+  }
+
+  // Store the time when the button was first pressed
+  uint32_t start_time = millis();
+  // While the button is being pressed
+  while (digitalRead(BTN_BUILTIN) == LOW) {
+    // Calculate the elapsed time
+    uint32_t elapsed_time = millis() - start_time;
+    // If the button has been pressed for less than 10 seconds, continue
+    if (elapsed_time < 10000u) {
+      yield();
+      continue;
+    }
+
+    // Blink the LED to indicate the start of the decommissioning process
+    for (uint8_t i = 0u; i < 10u; i++) {
+      digitalWrite(LED_BUILTIN, !(digitalRead(LED_BUILTIN)));
+      delay(100);
+    }
+
+    Serial.println("Starting decommissioning process, device will reboot...");
+    Serial.println();
+    digitalWrite(LED_BUILTIN, LED_BUILTIN_INACTIVE);
+    // This function will not return
+    // The device will restart once decommissioning has finished
+    Matter.decommission();
+  }
+}
+```
+
+This is a very simple code that consist of the Arduino basic sketch parts:
+- In the `setup()` function we initialize the Matter connectivity and the fan output pin. 
+- In the `loop()` function we listen to any fan control command sent from the smartphone or voice assistant and then adjust the fan speed setting the PWM signal accordingly.
+
+Once you uploaded the example code to the Nano Matter, open the Serial Monitor and reset the board.
+
+![QR Code URL](assets/serial-monitor-v4.png)
+
+There you will find the URL that generates the QR for the Matter device commissioning.
+
+### Commissioning 
```

Comment:
any way to improve this header? Device commissioning? we need a way to explain the less advanced users what is commissioning